crates/ruff_python_formatter/tests/snapshots/format@fmt_skip__compound_one_liners.py.snap

```diff
@@ -0,0 +1,289 @@
+
+@property
+def prop_method(self): return self._value  # fmt: skip
+
+# Class definitions on one line
+class SimpleClass: pass  # fmt: skip
+class GenericClass(Generic[T]): pass  # fmt: skip
+
+# Try/except blocks
+try: risky_operation()  # fmt: skip
+except ValueError: handle_error()  # fmt: skip
+except: handle_any_error()  # fmt: skip
+else: success_case()  # fmt: skip
+finally: cleanup()  # fmt: skip
+
+# Match statements (Python 3.10+)
+match value:
+    case 1: print("one")  # fmt: skip
+    case _: print("other")  # fmt: skip
+
+# With statements
+with open("file.txt") as f: content = f.read()  # fmt: skip
+with context_manager() as cm: result = cm.process()  # fmt: skip
+
+# Async variants
+async def async_func(): return await some_call()  # fmt: skip
+async for item in async_iterator(): await process(item)  # fmt: skip
+async with async_context() as ctx: await ctx.work()  # fmt: skip
+
+# Complex expressions that would normally format
+def complex_expr(): return [x for x in range(100) if x % 2 == 0 and x > 50]  # fmt: skip
+if condition_a and condition_b or (condition_c and not condition_d): execute_complex_logic()  # fmt: skip
+
+# Edge case: comment positioning
+def func_with_comment():  # some comment
+    return "value"  # fmt: skip
+
+# Edge case: multiple fmt: skip (only last one should matter)
+def multiple_skip(): return "test"  # fmt: skip  # fmt: skip
+
+# Should NOT be affected (already multiline)
+def multiline_func():
+    return "this should format normally"
+
+if long_condition_that_spans \
+   and continues_on_next_line:
+    print("multiline condition")
+
+# Mix of skipped and non-skipped
+for i in range(10): print(f"item {i}")  # fmt: skip
+for j in range(5):
+    print(f"formatted item {j}")
+
+# With trailing comma that would normally be removed
+def trailing_comma_func(a, b, c,): return a + b + c  # fmt: skip
+
+# Dictionary/list comprehensions
+def dict_comp(): return {k: v for k, v in items.items() if v is not None}  # fmt: skip
+def list_comp(): return [x * 2 for x in numbers if x > threshold_value]  # fmt: skip
+
+# Lambda in one-liner
+def with_lambda(): return lambda x, y, z: x + y + z if all([x, y, z]) else None  # fmt: skip
+
+# String formatting that would normally be reformatted
+def format_string(): return f"Hello {name}, you have {count} items in your cart totaling ${total:.2f}"  # fmt: skip
+
+# loop else clauses
+for i in range(2): print(i) # fmt: skip
+else: print("this") # fmt: skip
+
+
+while foo(): print(i) # fmt: skip
+else: print("this") # fmt: skip
+
+# again but only the first skip
+for i in range(2): print(i) # fmt: skip
+else: print("this")
+
+
+while foo(): print(i) # fmt: skip
+else: print("this")
+
+# again but only the second skip
+for i in range(2): print(i)
+else: print("this") # fmt: skip
+
+
+while foo(): print(i)
+else: print("this") # fmt: skip
+
+# multiple statements in body
+if True: print("this"); print("that") # fmt: skip
+```
```

Comment:
What about:

```py
if (
	long_condition
):  a     +     b # fmt: skip
```


I would expect that this doesn't get formatted according to how `fmt:skip`s semantic is defined.

I don't expect this to be hard to support. It might be as easy as only changing the `countains_line_break` check to only test from `:` to the end of the clause